torch/_inductor/fx_passes/fsdp.py

```diff
@@ -0,0 +1,347 @@
+import logging
```

Comment:
hmm if we want to land this as an initial port just so we have something to start iterating on i'm ok with that, but my vote would be to not hardcode FSDP in the file / pass names. Since theoretically this should be a generic pass for doing bucketing (and when we want to later do bucketing for allreduces, we should hopefully be able to re-use a big chunk of the code).

What do folks think of renaming this to `bucketing.py` or something? (and calling this a generic allgather_bucketing pass)